noxfile.py

```diff
@@ -180,3 +181,64 @@ def make(session: nox.Session):
         *(args.make_args or ("clean", "coredocs")),
     ]
     session.run("make", "-C", "docs/docsite", *make_args, external=True)
+
+
+def _rm_tmpdir(tmpdir):
+    rmtree(tmpdir, ignore_errors=True)
+
+
+@nox.session
+def tag(session: nox.Session):
+    """
+    Check the core repo for new releases and create tags in ansible-documentation
+    """
+    install(session, req="tag")
+
+    CORE_REPO = "https://github.com/ansible/ansible"
+    DOCS_REPO = "https://github.com/ansible/ansible-documentation"
+
+    args = list(session.posargs)
+    tmpdir = None
+
+    if not any(arg.startswith(("--core", "--docs")) for arg in args):
+        if not which("git"):
+            session.error("git is not installed or not found int PATH")
+
+        tmpdir = session.create_tmp()
+        core_dir = "".join((str(tmpdir), "/ansible"))
+        docs_dir = "".join((str(tmpdir), "/ansible-docs"))
+
+        try:
+            session.run(
+                "git",
+                "clone",
+                "--quiet",
+                CORE_REPO,
+                core_dir,
+                external=True,
+            )
+            args.extend(["--core", core_dir])
+        except Exception as e:
+            session.error(f"Could not update core repository: {e}")
+        try:
+            session.run(
+                "git",
+                "clone",
+                "--quiet",
+                DOCS_REPO,
+                docs_dir,
+                external=True,
+            )
+            args.extend(["--core", docs_dir])
+        except Exception as e:
+            session.error(f"Could not update docs repository: {e}")
```

Comment:
This is inefficient, as it has to re-clone the entire repository each time. I'd prefer if we kept this barebones and let the user handle the clones themselves.

Reply:
That's contrary to the design intention I expressed [above](https://github.com/ansible/ansible-documentation/pull/1649#discussion_r1654603061). If "this should just work" isn't the desired approach, I'll need to go back and redesign things. But, I still hold that running `nox -s tag` shouldn't require pre-setup or knowledge to work out-of-the-box. If it does, we're not that much further along than the current state.

Reply:
I apologize for not communicating this clearly earlier, but the intention of the script I wrote (or a nox session to wrap it that I suggested previously) was never to automatically manage clones (creating new ones, determining which remote was the upstream vs. a fork on an existing clone, or anything like that). All I wanted was for a relatively straightforward and maintainable script that would operate on a developer's existing docs and core clones. Unlike the other nox sessions which are meant for wider use by contributors and should be more fool-proof, this script will only be used by seasoned contributors with commit rights who should already checkouts that are properly set up. The other maintainers may think differently (please speak up if you do :), but this is my point of view as the script's author. I am happy to go through and make suggestions about Python code style or creating persistent clones in a `gitignore`ed subdirectory to make this more efficient, but my current inclination is to keep this as simple as possible and simply run the existing script in its own venv.

Reply:
I agree with @gotmax23 here.

Reply:
@gotmax23 So what you're really looking for for this `nox` session is dependency isolation (instead of having to install the reqs file), and then a straight pass through of everything else? 

I very much didn't get that based on how everything else is setup to work, but I can go ahead and reduce this down to that. 

Personally, I'd also be interested in investigating the other route with clones etc, but not if its not going to be used.